src/foam/nanos/auth/PermissionedPropertyDAO.js

```diff
@@ -127,26 +127,28 @@ foam.CLASS({
   FObject obj = oldObj.fclone();
   AuthService auth = (AuthService) x.get("auth");
 
-  if ( propertyMap_.containsKey(of) ) {
-    List properties = propertyMap_.get(of);
-    Iterator e = properties.iterator();
-    while ( e.hasNext() ) {
-      PropertyInfo axiom = (PropertyInfo) e.next();
-      maybeRemove(axiom, of, auth, x, obj, oldObj);
-    }
-  } else {
+  if ( ! propertyMap_.containsKey(of) ) {
     List<PropertyInfo> properties = new ArrayList<>();
     List list = oldObj.getClassInfo().getAxiomsByClass(PropertyInfo.class);
     Iterator e = list.iterator();
     while ( e.hasNext() ) {
       PropertyInfo axiom = (PropertyInfo) e.next();
       if ( axiom.getPermissionRequired() ) {
-        properties.add(axiom);
-        maybeRemove(axiom, of,auth, x, obj, oldObj);
+        boolean authCheck = auth.check(x, of + ".rw." + axiom.getName().toLowerCase()) || auth.check(x, of + ".ro." + axiom.getName().toLowerCase());
+        if ( ! authCheck ) {
```

Comment:
This doesn't make sense. You're only doing an auth check on the first user and putting the axioms in the map based on whether that particular user has permission or not. It needs to do the auth check every time because different users will have different permissions.